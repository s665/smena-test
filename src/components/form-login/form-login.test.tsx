import React from 'react';
import renderer from 'react-test-renderer';
import FormLogin from "./form-login";
import {BrowserRouter} from 'react-router-dom'

describe('render form-login', () => {
    it('render error', () => {
        const three = renderer
            .create(
                <BrowserRouter>
                    <FormLogin errorMessage="Произошла ошибка"
                               onFormSubmit={() => null}/>
                </BrowserRouter>
            )
            .toJSON()
        expect(three).toMatchSnapshot()
    })
    it('render no error', () => {
        const three = renderer
            .create(
                <BrowserRouter>
                    <FormLogin onFormSubmit={() => null}/>
                </BrowserRouter>
            )
            .toJSON()
        expect(three).toMatchSnapshot()
    })
})
