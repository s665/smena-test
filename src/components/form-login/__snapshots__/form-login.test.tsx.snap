// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render form-login render error 1`] = `
<div
  className="ant-card ant-card-bordered"
  style={
    Object {
      "margin": "0 15px",
      "width": "500px",
    }
  }
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center ant-divider-plain"
      role="separator"
    >
      <span
        className="ant-divider-inner-text"
      >
        <h2>
          Вход
        </h2>
      </span>
    </div>
    <form
      className="ant-form ant-form-vertical"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-label"
          style={Object {}}
        >
          <label
            className="ant-form-item-required"
            htmlFor="username"
            title="Логин"
          >
            Логин
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-input-affix-wrapper"
                onMouseUp={[Function]}
              >
                <span
                  className="ant-input-prefix"
                >
                  <span
                    aria-label="user"
                    className="anticon anticon-user"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="user"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  className="ant-input"
                  id="username"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  style={null}
                  type="text"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-label"
          style={Object {}}
        >
          <label
            className="ant-form-item-required"
            htmlFor="password"
            title="Пароль"
          >
            Пароль
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-input-affix-wrapper ant-input-password"
                onMouseUp={[Function]}
              >
                <span
                  className="ant-input-prefix"
                >
                  <span
                    aria-label="lock"
                    className="anticon anticon-lock"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="lock"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  action="click"
                  className="ant-input"
                  id="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  style={null}
                  type="password"
                  value=""
                />
                <span
                  className="ant-input-suffix"
                >
                  <span
                    aria-label="eye-invisible"
                    className="anticon anticon-eye-invisible ant-input-password-icon"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseUp={[Function]}
                    role="img"
                    tabIndex={-1}
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye-invisible"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                      />
                      <path
                        d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-alert ant-alert-error"
        data-show={true}
        role="alert"
        style={
          Object {
            "marginBottom": "15px",
          }
        }
      >
        <span
          aria-label="close-circle"
          className="anticon anticon-close-circle ant-alert-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
            />
          </svg>
        </span>
        <div
          className="ant-alert-content"
        >
          <div
            className="ant-alert-message"
          >
            Произошла ошибка
          </div>
          <div
            className="ant-alert-description"
          />
        </div>
        <button
          className="ant-alert-close-icon"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            aria-label="close"
            className="anticon anticon-close"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Войти
                </span>
              </button>
              <a
                href="/register/"
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": "15px",
                  }
                }
              >
                Регистрация
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`render form-login render no error 1`] = `
<div
  className="ant-card ant-card-bordered"
  style={
    Object {
      "margin": "0 15px",
      "width": "500px",
    }
  }
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center ant-divider-plain"
      role="separator"
    >
      <span
        className="ant-divider-inner-text"
      >
        <h2>
          Вход
        </h2>
      </span>
    </div>
    <form
      className="ant-form ant-form-vertical"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-label"
          style={Object {}}
        >
          <label
            className="ant-form-item-required"
            htmlFor="username"
            title="Логин"
          >
            Логин
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-input-affix-wrapper"
                onMouseUp={[Function]}
              >
                <span
                  className="ant-input-prefix"
                >
                  <span
                    aria-label="user"
                    className="anticon anticon-user"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="user"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  className="ant-input"
                  id="username"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  style={null}
                  type="text"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-label"
          style={Object {}}
        >
          <label
            className="ant-form-item-required"
            htmlFor="password"
            title="Пароль"
          >
            Пароль
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-input-affix-wrapper ant-input-password"
                onMouseUp={[Function]}
              >
                <span
                  className="ant-input-prefix"
                >
                  <span
                    aria-label="lock"
                    className="anticon anticon-lock"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="lock"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  action="click"
                  className="ant-input"
                  id="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  style={null}
                  type="password"
                  value=""
                />
                <span
                  className="ant-input-suffix"
                >
                  <span
                    aria-label="eye-invisible"
                    className="anticon anticon-eye-invisible ant-input-password-icon"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseUp={[Function]}
                    role="img"
                    tabIndex={-1}
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye-invisible"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                      />
                      <path
                        d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Войти
                </span>
              </button>
              <a
                href="/register/"
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": "15px",
                  }
                }
              >
                Регистрация
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
